backend/regalloc/regalloc_split_state.ml

```diff
@@ -106,6 +107,139 @@ end = struct
     !destructions_at_end, definitions_at_beginning
 end
 
+(* Optimizes `destructions_at_end` by avoiding the repeated spill of a constant.
+   If a register is set only once, then all spills of that register dominated by
+   a spill of the same register can be deleted. *)
+module RemoveDominatedSpillsForConstants : sig
+  val optimize :
+    Cfg_with_liveness.t ->
+    Cfg_dominators.t ->
+    destructions_at_end:destructions_at_end ->
+    destructions_at_end
+end = struct
+  (* CR-someday xclerc for xclerc: move to Cfg_loop_infos? *)
+  let is_in_loop : Cfg_loop_infos.t -> Label.t -> bool =
+   fun loops label ->
+    Cfg_loop_infos.EdgeMap.exists
+      (fun _ (loop : Cfg_loop_infos.loop) -> Label.Set.mem label loop)
+      loops.loops
+
+  type set =
+    | Exactly_once
+    | At_least_twice
+
+  let string_of_set = function
+    | Exactly_once -> "exactly once"
+    | At_least_twice -> "at least twice"
+
+  let rec remove_dominated_spills :
+      Cfg_dominators.t ->
+      Cfg_loop_infos.t ->
+      Cfg_dominators.dominator_tree ->
+      num_sets:set Reg.Tbl.t ->
+      already_spilled:Label.t Reg.Map.t ->
+      destructions_at_end:destructions_at_end ->
+      destructions_at_end =
+   fun doms loops tree ~num_sets ~already_spilled ~destructions_at_end ->
```

Comment:
I don't see how `loop` argument is used.

Reply:
`loops` is used when calling `is_in_loop`
below; it is actually necessary: if there is
only one set but in a loop then it is not
safe to apply the optimization.

Reply:
my comment was about `remove_dominated_spills` function, it does not use `loops`.

Reply:
Ah, sorry - deleted.